akka-remote/src/main/scala/akka/remote/artery/tcp/SecureRandomFactory.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.remote.artery.tcp
+
+import java.security.SecureRandom
+
+import akka.annotation.InternalApi
+import akka.event.LogMarker
+import akka.event.MarkerLoggingAdapter
+import com.typesafe.config.Config
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object SecureRandomFactory {
+
+  val GeneratorSha1Prng = "SHA1PRNG"
+  val GeneratorNativePrng = "NativePRNG"
+  val GeneratorJdkSecureRandom = "SecureRandom"
+
+  /**
+   * INTERNAL API
+   */
+  @InternalApi
+  // extracted as a method for testing
+  private[tcp] def rngConfig(config: Config) = {
+    config.getString("random-number-generator")
+  }
+
+  def createSecureRandom(config: Config, log: MarkerLoggingAdapter): SecureRandom = {
+    createSecureRandom(rngConfig(config), log)
+  }
+
+  def createSecureRandom(randomNumberGenerator: String, log: MarkerLoggingAdapter): SecureRandom = {
+    val rng = randomNumberGenerator match {
+      case s @ (GeneratorSha1Prng | GeneratorNativePrng) =>
+        log.debug("SSL random number generator set to: {}", s)
+        // SHA1PRNG needs /dev/urandom to be the source on Linux to prevent problems with /dev/random blocking
+        // However, this also makes the seed source insecure as the seed is reused to avoid blocking (not a problem on FreeBSD).
+        SecureRandom.getInstance(s)
+
+      case "" | GeneratorJdkSecureRandom =>
+        log.debug("SSL random number generator set to [SecureRandom]")
+        new SecureRandom
+
+      case unknown =>
+        log.warning(
+          LogMarker.Security,
+          "Unknown SSL random number generator [{}] falling back to SecureRandom",
+          unknown)
+        new SecureRandom
```

Comment:
I don't see any reason why we wouldn't just let users select whatever algorithm they want.

```suggestion
      case "" | GeneratorJdkSecureRandom =>
        log.debug("Using platform default SecureRandom algorithm for SSL")
        new SecureRandom
      case custom =>
        log.debug("Using {} SecureRandom algorithm for SSL", custom)
        SecureRandom.getInstance(custom)
```

If this suggestion is accepted, then the `GeneratorSha1Prng` and `NativePRNG` constants above sohuld also be deleted.